gaia-fast-list.js

```diff
@@ -772,6 +782,42 @@ Internal.prototype = {
       : '0';
   },
 
+  populateItemDetail(el, i) {
+    if (!this.getItemImageSrc) return;
+
+    var img = this.getItemImage(el);
+    if (!img) return;
+
+    var record = this.getRecordAt(i);
+    Promise.resolve(this.getItemImageSrc(record, i))
+      .then(src => {
+
+        // There is a chance that the item
+        // could have been recycled before
+        // the user was able to fetch the image.
+        // Abort here if that's the case.
+        if (el.dataset.index != i) return debug('item recycled');
+
+        img.src = src;
+        img.onload = () => {
+          img.style.transition = 'opacity 250ms';
+          img.style.opacity = 1;
```

Comment:
I thought setting style properties directly could be a small perf win over setting a class.